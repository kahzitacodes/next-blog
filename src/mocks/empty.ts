import { EmptyProps } from '@/components/empty'

export const emptyMock: EmptyProps = {
  title: 'Oops!',
  description: 'Page not found'
}
