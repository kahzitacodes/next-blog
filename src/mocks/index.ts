export * from './navigation'
export * from './profile'
export * from './social-links'
export * from './post'
export * from './pre'
export * from './pagination'
