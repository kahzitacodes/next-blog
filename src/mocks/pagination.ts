export const paginationMock = {
  currentPage: 1,
  totalPages: 10
}
