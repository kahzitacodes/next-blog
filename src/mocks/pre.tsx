import { PreProps } from '@/components/pre'

export const preMock: PreProps = {
  children: <h1>Title</h1>
}
