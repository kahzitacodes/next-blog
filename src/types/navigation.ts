import * as LucideIcons from 'lucide-react'

export type NavItem = {
  title: string
  href: string
  icon?: keyof typeof LucideIcons
}
