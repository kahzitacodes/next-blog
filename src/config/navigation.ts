import { NavItem } from '@/types'

export const navigationConfig: NavItem[] = [
  {
    title: 'Home',
    href: '/'
  },
  {
    title: 'Posts',
    href: '/posts'
  },
  {
    title: 'About me',
    href: '/about'
  }
]
