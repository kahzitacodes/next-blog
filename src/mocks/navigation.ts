import { NavItem } from '@/types'

export const navigationMock: NavItem[] = [
  {
    title: 'Home',
    href: '/'
  },
  {
    title: 'About',
    href: '/about'
  },
  {
    title: 'Contact',
    href: '/contact'
  }
]
