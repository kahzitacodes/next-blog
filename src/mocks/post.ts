import { Post } from '@/types'

export const postMock: Post = {
  $frontMatter: {
    date: '2025-04-04',
    description: 'Description',
    image: '/assets/images/image.png',
    title: 'Post title',
    tags: ['react']
  },
  $readingTime: 3,
  body: {
    code: '',
    raw: ''
  },
  id: '1',
  slug: 'post-3'
}
