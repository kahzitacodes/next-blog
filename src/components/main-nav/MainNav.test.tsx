import { render, screen } from '@testing-library/react'
import { navigationMock } from '@/mocks/navigation'
import { MainNav } from '.'

describe('<MainNav />', () => {
  it('should render successfully', () => {
    const { container } = render(<MainNav items={navigationMock} />)

    expect(screen.getByText(/home/i)).toBeInTheDocument()
    expect(screen.getByText(/about/i)).toBeInTheDocument()
    expect(screen.getByText(/contact/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
