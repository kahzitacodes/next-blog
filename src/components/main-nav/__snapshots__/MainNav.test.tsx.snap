// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainNav /> should render successfully 1`] = `
<nav
  class="items-center justify-between hidden lg:flex"
>
  <ul
    class="flex gap-6 font-medium flex-row"
    direction="horizontal"
  >
    <li
      class="flex gap-2 relative [&>a]:hover:text-link [&>a]:transition-colors [&>a]:duration-200 [&>svg]:hover:text-link"
    >
      <a
        class="flex justify-center gap-3"
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="flex gap-2 relative [&>a]:hover:text-link [&>a]:transition-colors [&>a]:duration-200 [&>svg]:hover:text-link"
    >
      <a
        class="flex justify-center gap-3"
        href="/about"
      >
        About
      </a>
    </li>
    <li
      class="flex gap-2 relative [&>a]:hover:text-link [&>a]:transition-colors [&>a]:duration-200 [&>svg]:hover:text-link"
    >
      <a
        class="flex justify-center gap-3"
        href="/contact"
      >
        Contact
      </a>
    </li>
  </ul>
</nav>
`;
